File: napakalaki/src/Napakalaki/SpecificBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Napakalaki;

import java.util.ArrayList;

/**
 *
 * @author Marcos
 */
public class SpecificBadConsequence extends BadConsequence{
    
    private ArrayList<TreasureKind> specificHiddenTreasure= new ArrayList();
    private ArrayList<TreasureKind> specificVisibleTreasure= new ArrayList();
    
    public SpecificBadConsequence(String _text, int _levels, ArrayList<TreasureKind> tVisible, ArrayList<TreasureKind> tHidden){
        super(_text, _levels);
        specificHiddenTreasure=tVisible;
        specificVisibleTreasure=tHidden;
    }
    
    public ArrayList<TreasureKind> getTesorosEspecificosVisible(){
        return specificVisibleTreasure;
    }
    
    public ArrayList<TreasureKind> getTesorosEspecificosOcultos(){
        return specificHiddenTreasure;
    }
    
    @Override
    public boolean isEmpty(){
       if(specificVisibleTreasure.isEmpty() && specificHiddenTreasure.isEmpty()){
           return true;
       }
       else {
           return false;
       }
    }
    
    @Override
    public void substractVisibleTreasure(Treasure t){
        if(!specificVisibleTreasure.isEmpty()){
            specificVisibleTreasure.remove(t.getType());
        }
    }
    
    @Override
    public void substractHiddenTreasure(Treasure t){
        if(specificHiddenTreasure.isEmpty()){
            specificHiddenTreasure.remove(t.getType());
        }
    }
    
    @Override
    public SpecificBadConsequence adjustToFitTreasureList(ArrayList<Treasure> v, ArrayList<Treasure> h){
       SpecificBadConsequence resultado = new SpecificBadConsequence(super.getTtext(), super.getLevels(), new ArrayList(), new ArrayList());
       if(this.specificHiddenTreasure.size()>0){
            for(int i=0; i<this.specificHiddenTreasure.size(); i++){
             for(int j=0;j<h.size();j++){
                 if(this.specificHiddenTreasure.get(i)==h.get(j).getType()){
                       resultado.specificHiddenTreasure.add(this.specificHiddenTreasure.get(i));
                    }
             }
            }
        }
        if(this.specificVisibleTreasure.size()>0){
         for(int i=0;i<this.specificVisibleTreasure.size();i++){
             for(int j=0;j<v.size();j++){
                  if(this.specificVisibleTreasure.get(i)==v.get(j).getType()){
                     resultado.specificVisibleTreasure.add(this.specificVisibleTreasure.get(i));
                    }
             }
            }
        }
        return resultado;
    }
    
    @Override
    public String toString(){
        return "Texto = " + (super.getTtext())
                + " Levels = " + Integer.toString(super.getLevels())
                + "spedificHiddenTreasures = " + specificHiddenTreasure
                + " spedificVisibleTreasure = " + specificVisibleTreasure
                + "\n";
    }
}

File: napakalaki(ruby)/lib/specific_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame
  require_relative 'bad_consequence'
class SpecificBadConsequence < BadConsequence
  
  attr_accessor :specificVisibleTreasure, :specificHiddenTreasure
  
  def initialize(text, levels, v, h)
    super(text,levels)
    @specificVisibleTreasure=v
    @specificHiddenTreasure=h
  end
  
  def isEmpty
    if @specificVisibleTreasure.empty? && @specificHiddenTreasure.empty?
      return true;
    else 
      return false
    end
  end
  
  def substractVisibleTreasure(t)
    if !@specificVisibleTreasure.empty?
      @specificVisibleTreasure.delete(t.type)
    end
  end
  
  def substractHiddenTreasure(t)
    if !@specificHiddenTreasure.empty?
      @specificHiddenTreasure.delete(t.type)
    end
  end
  
  def adjustToFitTreasureLists(v, h)
    resultado = SpecificBadConsequence.new(self.text, self.levels, Array.new, Array.new)
    if @specificHiddenTreasure.length>0
      @specificHiddenTreasure.each do |x|
        h.each do |t|
          if x==t.type
            resultado.specificHiddenTreasure.push(x)
          end
        end
      end
    end
    if @specificVisibleTreasure.length>0
      @specificVisibleTreasure.each do |sv|
        v.each do |x|
          if sv==x.type
            resultado.specificVisibleTreasure.push(sv)
          end
        end
      end
    end
    return resultado;
  end
  
  def to_s()
   " \n #{super.to_s} +  \n Tesoros Visibles perdidos: #{@specificVisibleTreasure} \n Tesoros Ocultos perdidos: #{@specificHiddenTreasure}"
  end
   
  public_class_method :new
end
end